Tests/IntegrationTests.Shared/LifetimeTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using NUnit.Framework;
+using Realms;
+
+namespace IntegrationTests.Shared
+{
+    [TestFixture, Preserve(AllMembers = true)]
+    public class LifetimeTests
+    {
+        // This method was extracted to ensure that the actual realm instance
+        // isn't preserved in the scope of the test, even when the debugger is running.
+        private WeakReference GetWeakRealm()
+        {
+            return new WeakReference(Realm.GetInstance("LifetimeTests.realm"));
+        }
+
+        [Test]
+        public void RealmObjectsShouldKeepRealmAlive()
+        {
+            // Arrange
+            var realm = GetWeakRealm();
+            Person person = null;
+            ((Realm)realm.Target).Write(() => { person = ((Realm)realm.Target).CreateObject<Person>(); });
+
+            // Act
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+
+            // Assert
+            Assert.That(realm.IsAlive);
+            Assert.That(((Realm)realm.Target).IsClosed, Is.False);
+            Assert.That(person.IsValid);
+        }
+
+        [Test]
+        public void FinalizedRealmsShouldNotInvalidateSiblingRealms()
```

Comment:
Should we allow people to have a realm instance in a `using` statement and another realm on the same file, in the same thread, and expect the second to remain open?